### PR TITLE
EIP-1344: Add chain id opcode

### PR DESCRIPTION
**Pull request by rmeissner**
PR based on the issue #901 

Discussions at https://ethereum-magicians.org/t/add-chain-id-opcode-for-replay-protection-when-handling-signed-messages-in-contracts/1131

### CODE REVIEW

File: EIPS/eip-1344.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+---
+eip: 1344
+title: ChainID opcode
+author: Richard Meissner (@rmeissner)
+category: Core
+type: Standards Track
+status: Draft
+created: 2018-08-22
+---
```
Comment (Arachnid) [inline EIPS/eip-1344.md]: Please add a discussions-to URL.

**Comment by rmeissner** [comment]
@Arachnid Hey, what do I need to do to get this merged? :D

**Comment by fubuloubu** [comment]
Wanted to ping this EIP. I think inclusion of this opcode for Istanbul may be very valuable for EIP-712 applications (being able to programmatically query chain ID for constructing the domain separator)

**Comment by rmeissner** [comment]
Would love that too, still not sure what the best way is to push this forward :/

**Comment by fubuloubu** [comment]
Have to go through the process, and place it on the Istanbul roadmap before May 17th.

Would love to help you out in making this happen, if another similar EIP hasn't already been planned.

**Comment by fubuloubu** [comment]
@rmeissner picking up the conversation on EthMagicians, probably needs some additional detail as it's missing the "Rationale" and "Backwards Compatibility" sections. Not sure if we should write Test Cases, but the Implementation is easier enough it is probably okay leaving that empty.

**Pull request merged**